### PR TITLE
fix(application-config): allow to define `entryPointUriPath` with env placeholder

### LINKED ISSUE #2470: Can't use env var for entryPointUriPath?
The docs don't say you can't do that, but I'm hitting an error when trying to run the app locally. It's kinda necessary for our use case to have dev and prod versions of our custom app. We're using env vars for production.applicationId and production.url and I'm not seeing issues with those yet.

Screen Shot 2022-02-04 at 10 04 07 AM

### PR DESCRIPTION
**Pull request by CarlosCortizasCT**
fixes #2470

### CODE REVIEW

**Review by Rhotimee — APPROVED** [review]

File: .changeset/slimy-parrots-deny.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@commercetools-frontend/application-config': minor
+---
+
+Allow env vars in entryPointUriPath app config at schema level
```
Comment (emmenko) [inline .changeset/slimy-parrots-deny.md:5]:
```suggestion
Allow to use environment variable placeholders for the `entryPointUriPath` field.
```

File: .changeset/slimy-parrots-deny.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@commercetools-frontend/application-config': minor
```
Comment (emmenko) [inline .changeset/slimy-parrots-deny.md:2]: I think this is more of a bug

```suggestion
'@commercetools-frontend/application-config': patch
```

File: packages/application-config/schema.json [hunk 3]
```diff
@@ -28,7 +28,14 @@
     "entryPointUriPath": {
       "description": "See https://docs.commercetools.com/custom-applications/api-reference/application-config#entrypointuripath",
       "type": "string",
```
Comment (emmenko) [inline packages/application-config/schema.json:30]: Move this into the "pattern" object.

File: packages/application-config/schema.json [hunk 4]
```diff
@@ -28,7 +28,14 @@
...
-      "pattern": "^[^\\-_]([0-9a-z]|[\\-_](?![\\-_])){2,64}[^\\-_]$"
+      "oneOf": [
+        {
+          "pattern": "^[^\\-_]([0-9a-z]|[\\-_](?![\\-_])){2,64}[^\\-_]$"
```
Comment (emmenko) [inline packages/application-config/schema.json:33]:
```suggestion
        {
          "type": "string",
          "pattern": "^[^\\-_]([0-9a-z]|[\\-_](?![\\-_])){2,64}[^\\-_]$"
```

File: packages/application-config/src/schema.ts [hunk 5]
```diff
@@ -17,7 +17,13 @@ export interface JSONSchemaForCustomApplicationConfigurationFiles {
...
    * See https://docs.commercetools.com/custom-applications/api-reference/application-config#entrypointuripath
    */
-  entryPointUriPath: string;
+  entryPointUriPath: (
+    | {
+        [k: string]: unknown;
+      }
+    | EnvVariablePlaceholder
+  ) &
+    string;
```
Comment (emmenko) [inline packages/application-config/src/schema.ts:26]: This looks wrong. See other comment above.

**Review by emmenko — CHANGES_REQUESTED** [review]
Thanks. Can you also adjust the tests? I think you can use the `test/fixtures/config-env-variables.json` file to use the variable placeholder.

File: packages/application-config/src/schema.ts (hunk 5 above)
Comment (CarlosCortizasCT) [reply packages/application-config/src/schema.ts]: Updated the schema for a better types output with [this commit](https://github.com/commercetools/merchant-center-application-kit/pull/2474/commits/ffbac5c71ce9d58e56da2a892296c04c0355bf8e).

Thanks for the guidance.

**Review by emmenko — APPROVED** [review]

**Pull request merged**